src/dbnode/storage/mediator.go

```diff
@@ -33,15 +33,22 @@ import (
 	"go.uber.org/zap"
 )
 
-type mediatorState int
+type (
+	mediatorState            int
+	fileSystemProcessesState int
+)
 
 const (
-	fileOpCheckInterval = time.Second
-	tickCheckInterval   = 5 * time.Second
+	fileOpCheckInterval     = time.Second
+	tickCheckInterval       = 5 * time.Second
+	completionCheckInterval = 1 * time.Second
```

Comment:
Could even make this every 100ms since it only happens on close. I remember the kill signal only gives us 10s to exit so the faster the better I suppose.